### PR TITLE
[SPARK-40012][PYTHON][DOCS] Make pyspark.sql.dataframe examples self-contained

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR takes https://github.com/apache/spark/pull/37444 over with covering all examples in `pyspark.sql.dataframe`.

This PR proposes to improve the examples in `pyspark.sql.dataframe` by making each example self-contained with more realistic examples.

Closes #37444

### Why are the changes needed?

To make the documentation more readable and able to copy and paste directly in PySpark shell.

### Does this PR introduce _any_ user-facing change?

Yes, Documentation changes only

### How was this patch tested?

Manually ran each examples.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @zhengruifeng @viirya @xinrong-meng @Yikun @itholic @dcoliversun @khalidmammadov in case you find some time to take a look.

**Review by dcoliversun — COMMENTED** [review]
Looks pretty good to me. Heavy workload. I made some small suggestions, please correct me if I'm wrong.

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -3067,6 +3436,10 @@ def unionByName(self, other: "DataFrame", allowMissingColumns: bool = False) ->
         ----------
         other : :class:`DataFrame`
             Another :class:`DataFrame` that needs to be combined.
+        allowMissingColumns : bool, optional, default False
+           Specify whether to allow missing columns.
+
+           .. versionadded:: 3.1.0
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/sql/dataframe.py:3442]: This is actually intentional (versionadded is for the new parameter)

**Review by srowen — APPROVED** [review]

**Comment by Transurgeon** [comment]
@HyukjinKwon, thanks for taking this over.. I hope I helped you guys a bit atleast

**Comment by HyukjinKwon** [comment]
Thanks guys.

Merged to master.

**Comment by HyukjinKwon** [comment]
Sure, that was a big help @Transurgeon

**Pull request closed without merging**